lib/code_input.dart

```diff
@@ -227,8 +227,9 @@ class _CodeInputState extends State<CodeInput> {
                 final hasFocus = controller.selection.start == i;
                 final char = i < text.length ? text[i] : '';
                 final characterEntity = Padding(
-                    padding: widget.padding,
-                    child: widget.builder(hasFocus, char));
+                  padding: EdgeInsets.only(left: widget.spacing),
+                  child: widget.builder(hasFocus, char),
+                );
```

Comment:
On the first character, there shouldn't be any space on the left. So I'd change this to:

```
var characterWidget = widget.builder(hasFocus, char);
if (i > 0) {
  characterWidget = Padding(
    padding: EdgeInsets.only(left: widget.spacing),
    child: characterWidget,
  );
}
```

(Also changed `characterEntity` to `characterWidget`, I think that's a better name.)